src/yb/rocksdb/util/thread_posix.cc

```diff
@@ -243,7 +248,12 @@ int ThreadPool::UnSchedule(void* arg) {
 
 std::vector<yb::util::WaitStateInfoPtr> ThreadPool::GetThreadpoolWaitStates() {
   PthreadCall("lock", pthread_mutex_lock(&mu_));
-  std::vector<yb::util::WaitStateInfoPtr> res = bg_wait_states_;
+  std::vector<yb::util::WaitStateInfoPtr> res;
+  for (auto wait_state : bg_wait_states_) {
```

Comment:
Do we want to filter out the un-used ones, or keep them around?

Reply:
I think it's better to filter them out, otherwise we keep seeing some wait states with -2 as the query id and everything else is null